spec/binarySearchTree.js

```diff
@@ -0,0 +1,98 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import BinarySearchTree from '../src/BinarySearchTree'
+import BinaryNode from '../src/binarynode'
+
+chai.use(chaiChange)
+
+describe('BinarySearchTree', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(BinarySearchTree).to.be.a('function')
+  })
+
+  context('insert()', () => {
+    context('when the tree is empty', () => {
+      it('inserts a node with the specified value at the root', () => {
+          const myBST = new BinarySearchTree()
+
+          myBST.insert(4)
+          expect(myBST.root._value).to.be.equal(4)
```

Comment:
We should avoid testing the underlying implementation, using the API instead.  Having said that, I'm not sure how I would test this specific functionality without adding additional methods that aren't in spec (which isn't necessarily bad...):
```js
class Tree {
  getRootData() {
    return this.root.getData()
  }
}

class Node {
  getData() {
    return this._value
  }
}
```
Note that I also added the `getData` function to encapsulate the `_value` field.